### PR TITLE
A test for OSERDES routing

### PR DESCRIPTION
**Pull request by mkurc-ant**
This is a small test that checks SymbiFlow handling of OSERDES inputs. Demonstrates the https://github.com/SymbiFlow/symbiflow-arch-defs/issues/1258 issue.

This PR also updates pack patterns for OSERDES.

### CODE REVIEW

File: xc7/tests/oserdes_routing/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  INPUT_IO_FILE dummy.pcf
+  EXPLICIT_ADD_FILE_TARGET
+  )
+
+add_vivado_target(
+  NAME oserdes_routing_vivado
+  PARENT_NAME oserdes_routing
+  # The fasm diff test is disabled as it is expected to fail for now. Should
+  # be re-enabled when #1258 is fixed
+  DISABLE_DIFF_TEST
```
Comment (acomodi) [inline xc7/tests/oserdes_routing/CMakeLists.txt]: I believe the DIFF test should be enabled, as the purpose of this test is to verify this behavior. This way I could take over this PR which can be merged as soon as the const issue is fixed.

Comment (mkurc-ant) [reply xc7/tests/oserdes_routing/CMakeLists.txt]: Ok, I've enabled the diff test. Should fail in Vivado with net conflict between <const 0> and <const 1>

**Comment by acomodi** [comment]
@mkurc-ant I have pushed a fix to correctly calculate the ZINV and IS_INVERTED parameters.

**Comment by mkurc-ant** [comment]
The `oserdes_routing` diff fasm test passed CI:
```
[918/1107] cd /tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7-vendor/build/xc7/tests/oserdes_routing && diff -u /tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7-vendor/build/xc7/tests/oserdes_routing/oserdes_routing/artix7-xc7a50t-bottom-virt-xc7a50t-bottom-test/top.bit.fasm /tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7-vendor/build/xc7/tests/oserdes_routing/oserdes_routing/artix7-xc7a50t-bottom-virt-xc7a50t-bottom-test/design_oserdes_routing_vivado.bit.fasm
```

Failing tests are related to other issues:
1. Clock resource placement for the `iobuf_i2c` test:
```
ERROR: [Place 30-176] Unroutable Placement! The following clock source instance is driving the following locked load instances. The clock source instance is placed too far away from its load instance to be routable.
	CLK_HROW_TOP_R_X60Y78_BUFHCE_X1Y18_BUFHCE (BUFHCE.O) is locked to BUFHCE_X1Y18
```
2. And for the `murax` test:
```
ERROR: [DRC RTSTAT-5] Partial antennas: 3 net(s) have a partial antenna. The problem bus(es) and/or net(s) are 
```

I think we can merge the fix anyway.

**Pull request merged**